src/main/java/com/google/api/generator/gapic/composer/DefaultValueComposer.java

```diff
@@ -123,6 +134,13 @@ static Expr createDefaultValue(Field f, boolean useExplicitInitTypeInGenerics) {
     }
 
     if (f.type().equals(TypeNode.STRING)) {
+      if (useSampleCode
```

Comment:
Why not pass in the ResourceName-typed field instead if the field is a String? Then we don't need to change DefaultValueComposer at all.